automated install/basic-install.sh

```diff
@@ -93,7 +101,7 @@ if [ -z "${USER}" ]; then
     USER="$(id -un)"
 fi
 
-# whiptail dialog dimensions: 20 rows and 70 chars width assures to fit on small screens and is known to hold all content.
+# dialog dimensions: 20 rows and 70 chars width assures to fit on small screens and is known to hold all content.
 r=20
```

Comment:
There is a `--print-maxsize` command for `dialog` that may be useful here.

Reply:
You can use `r=-1` and  `c=-1` to fit the whole screen, but the blue background and title will be cover by the box.

Reply:
The option `--print-maxsize` will return the height and width values, like this:
```sh
root@pi4:~ $ dialog --print-maxsize
MaxSize: 30, 120
```

Reply:
When I aimed to fix the automatic max size detection some time ago, the agreed opinion was that the 70/20 dialogs, most users saw due to failing max size detection in `curl ... | bash` pipe, are actually more beautiful than large ones with plenty of empty space. So we went to use 70/20 hardcoded/aligned, regardless how the installer is invoked, including updates and reinstalls.

So I'd keep it that way, otherwise discuss whether to change to larger/dynamic dimensions in a dedicated issue/PR. And in case really max size is wanted, indeed `-1 -1` can be used, same as `0 0` in whiptail.

Reply:
I have plans for more PRs to work with the installer with the goal of POSIX compliance. (End goal is to be able to natively install on Alpine...)

The design decision for the one size boxes for everything was made when `whiptail` was first introduced to the script, some 5 years ago.

I'm open to letting the dialogs size themselves and if people are trying to run in a tiny tiny terminal window then they get what any other dialog-based install would show.

Reply:
Using `0 0` will auto size the boxes. `-1 -1` will maximize them.

Reply:
> I have plans for more PRs to work with the installer with the goal of POSIX compliance.

Sounds good. With POSIX you also mean to switch from `bash` to bourne shell?

> The design decision for the one size boxes for everything was made when `whiptail` was first introduced to the script, some 5 years ago.

It was implemented last year with: #4229
`stty size` just did not work when STDIN is a pipe, quite similar like now keyboard input does not work with `whiptail` 😞.

> if people are trying to run in a tiny tiny terminal window then they get what any other dialog-based install would show.

I recognised just now, that `dialog` automatically limits width and height to terminal dimensions, so on small screens, while the text may exceed the dialog, the boarders and buttons at least will not. `whiptail` doesn't do that, so on too small terminals, buttons may not be visible anymore. However, smaller than 70/20 should be pretty rare.

> Using `0 0` will auto size the boxes. `-1 -1` will maximize them.

Ah right, `-1 -1` isn't even supported by `whiptail` 😅. However, I personally don't like any of both: max size covers backtitle (as you noted already), text is not centred, being in the top-left corner only on large screens, and with auto-size, full width is used before a second line (in case no manual line breaks are used), leading to wide slim boxes.

Reply:
> It was implemented last year with: #4229

Ah yes, I should have said "The idea of the minimum size". 

> Sounds good. With POSIX you also mean to switch from `bash` to bourne shell?

Yep. `sh`|`dash`|`ash`

Reply:
Let's try `0 0`.

Reply:
I think all screens with just one line of text can use `0 0`.
Screens with more text (or receiving a different kind of input, like radio or checkbox) might need an adjustment.

Reply:
`0 0` looks bad. I'll leave it as is and it can be adjusted to use shell math to find a minimum.